pkg/antctl/raw/multicluster/common/common.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	multiclusterv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	multiclusterv1alpha2 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha2"
+	"antrea.io/antrea/pkg/antctl/raw"
+	multiclusterscheme "antrea.io/antrea/pkg/antctl/raw/multicluster/scheme"
+)
+
+func NewClient(cmd *cobra.Command) (client.Client, error) {
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return nil, err
+	}
+	restConfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restConfigTmpl)
+
+	k8sClient, err := client.New(restConfigTmpl, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return nil, err
+	}
+
+	return k8sClient, nil
+}
+
+func CreateClusterClaim(cmd *cobra.Command, k8sClient client.Client, namespace string, clusterset string, clusterID string, createdRes *[]*unstructured.Unstructured) error {
+	fmt.Fprintf(cmd.OutOrStdout(), "Creating ClusterClaim \"%s\" in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, namespace)
+	var createErr error
+	if createErr = k8sClient.Create(context.TODO(), newClusterClaim(clusterID, namespace, false)); createErr != nil {
+		if apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" already exists in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, namespace)
+			createErr = nil
+		} else {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ClusterClaim \"%s\": %v\n", multiclusterv1alpha2.WellKnownClusterClaimID, createErr)
+			return createErr
+		}
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" created in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, namespace)
+		*createdRes = append(*createdRes, &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "multicluster.crd.antrea.io/v1alpha2",
+				"kind":       "ClusterClaim",
+				"metadata": map[string]interface{}{
+					"name":      multiclusterv1alpha2.WellKnownClusterClaimID,
+					"namespace": namespace,
+				},
+			},
```

Comment:
I think you can assign `newClusterClaim(clusterID, namespace, false)` to a variable and convert it to interface object here. then you don't have to repeat this long structure.

Reply:
Done. Inorder to get the Kind and APIVersion. `TypeMeta` added in the function.